sympy/core/function.py

```diff
@@ -1180,7 +1179,6 @@ def __new__(cls, expr, *variables, **assumptions):
             else:
                 if not is_symbol:
                     new_v = Dummy('xi_%i' % i)
-                    new_v.dummy_index = hash(v)
```

Comment:
I have not found an example where this line seems to matter: its related to expressions like `f(g(x)).diff(g(x))` but its not clear to me how to hit this code twice (so that the hash matters.)